lib/presentation/detail_screen/culture_location/culture_location_info_view_model.dart

```diff
@@ -0,0 +1,64 @@
+import 'dart:convert';
+
+import 'package:flutter/material.dart';
+import 'package:kovel_app/domain/model/detail/culture_location/culture_location_detail.dart';
+import 'package:kovel_app/domain/model/detail/culture_location/culture_location_detail_info.dart';
+import 'package:kovel_app/domain/model/detail/tour_detail.dart';
+import 'package:kovel_app/domain/repository/tour_info_repository.dart';
+import 'package:kovel_app/presentation/components/icon_text_row.dart';
+
+class CultureLocationInfoViewModel with ChangeNotifier {
+  final TourInfoRepository _tourInfoRepository;
+
+  CultureLocationInfoViewModel({
+    required TourInfoRepository tourInfoRepository,
+  }) : _tourInfoRepository = tourInfoRepository;
+
+  bool _isLoading = true;
+  List<Widget> _widgets = [];
+  List<TourDetail> _tourDetailData = [];
+  List<CultureLocationDetail> _cultureLocationDetailData = [];
+  List<CultureLocationDetailInfo> _cultureLocationInfoData = [];
+
+  bool get isLoading => _isLoading;
+  List<Widget> get widgets => _widgets;
+  List<TourDetail> get tourDetailData => _tourDetailData;
+  List<CultureLocationDetail> get cultureLocationDetailData =>
+      _cultureLocationDetailData;
+  List<CultureLocationDetailInfo> get cultureLocationInfoData =>
+      _cultureLocationInfoData;
+
+  Future<void> getCultureLocationData(int id, int contentTypeId) async {
+    _isLoading = true;
+    notifyListeners();
+
+    _tourDetailData = await _tourInfoRepository.getDetailCommon(id: id);
+    _cultureLocationDetailData = await _tourInfoRepository
+        .getCultureLocationDetail(id: id, contentTypeId: contentTypeId);
+    _cultureLocationInfoData = await _tourInfoRepository
+        .getCultureLocationDetailInfo(id: id, contentTypeId: contentTypeId);
+
+    _isLoading = false;
+    showDetailData();
+    notifyListeners();
+  }
+
+  void showDetailData() async {
+    Map<String, dynamic> detailData =
+        jsonDecode(jsonEncode(_cultureLocationDetailData.first));
+
+    detailData.forEach((key, value) {
+      if (value != null &&
+          value != '' &&
+          key != 'contentId' &&
+          key != 'contentTypeId') {
+        _widgets.add(
+          IconTextRow(
+            icon: Icons.phone,
+            text: value.toString(),
+          ),
+        );
+      }
+    });
+  }
```

Comment:
이 부분 재밌게 잘 봤습니다!!  좋은 아이디어인 것 같습니다
그런데 toJson을 이용하면 jsonDecode, jsonEncodee 과정이 없어도 될 것 같습니다
```suggestion
void showDetailData() async {
    _cultureLocationDetailData.first.toJson().forEach((key, value) {
      if (value != null &&
          value != '' &&
          key != 'contentId' &&
          key != 'contentTypeId') {
        _widgets.add(
          IconTextRow(
            icon: Icons.phone,
            text: value.toString(),
          ),
        );
      }
    });
  }
```

Reply:
너무 좋은 코드여서 바로 커밋눌러버렷습니다 ㅋㅋ